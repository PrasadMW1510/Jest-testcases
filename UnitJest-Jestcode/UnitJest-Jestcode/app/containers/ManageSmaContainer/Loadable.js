/**
 *
 * Asynchronously loads the component for ManageSmaContainer
 *
 */

import Loadable from 'react-loadable';

export default Loadable({
  loader: () => import('./index'),
  loading: () => null,
});
