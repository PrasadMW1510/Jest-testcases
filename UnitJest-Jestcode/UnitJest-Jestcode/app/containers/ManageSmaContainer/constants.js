/*
 *
 * ManageSmaContainer constants
 *
 */

export const GET_MEDIA_SERVERS_REQUEST = 'ManageSmaContainer/GET_MEDIA_SERVERS_REQUEST';
export const GET_MEDIA_SERVERS_REQUEST_SUCCESS =
  'ManageSmaContainer/GET_MEDIA_SERVERS_REQUEST_SUCCESS';
export const GET_MEDIA_SERVERS_REQUEST_FAILURE =
  'ManageSmaContainer/GET_MEDIA_SERVERS_REQUEST_FAILURE';
