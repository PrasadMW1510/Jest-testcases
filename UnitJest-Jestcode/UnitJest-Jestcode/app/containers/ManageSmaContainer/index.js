/**
 *
 * ManageSmaContainer
 *
 */

export { default } from './ManageSmaContainer';
