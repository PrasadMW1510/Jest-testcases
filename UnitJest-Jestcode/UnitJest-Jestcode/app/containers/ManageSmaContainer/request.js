import API from 'utils/request';

/**
 * Get media servers for the given district
 * @param sessionId
 * @param districtId
 */
export const getMediaServers = (sessionId, districtId) =>
  API.get('/SlmsMediaServer', {
    params: {
      command: 'getMediaServers',
      sid: sessionId,
      district_id: districtId,
    },
  }).then(response => response.output_data[0].media_servers[0].media_server);
