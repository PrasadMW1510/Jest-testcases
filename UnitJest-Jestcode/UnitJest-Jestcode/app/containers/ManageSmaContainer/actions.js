import { generalFailure } from 'containers/App/actions';
import * as Constants from './constants';

/**
 * Handles API logic for making calls needed for
 * getting media servers
 * @returns {{type}}
 */
export function getMediaServersRequest() {
  return {
    type: Constants.GET_MEDIA_SERVERS_REQUEST,
  };
}

/**
 * Handles successful fetch of media servers
 * @returns {{type}}
 */
export function handleMediaServersRequestSuccess(mediaServers) {
  return {
    mediaServers,
    type: Constants.GET_MEDIA_SERVERS_REQUEST_SUCCESS,
  };
}

/**
 * Handles API logic for making calls needed for
 * Add Edit Teacher Modal are a failure
 *
 * @param error
 * @returns {*}
 */
export function handleMediaServersRequestFailure(error) {
  return generalFailure({
    type: Constants.GET_MEDIA_SERVERS_REQUEST_FAILURE,
    error,
  });
}
