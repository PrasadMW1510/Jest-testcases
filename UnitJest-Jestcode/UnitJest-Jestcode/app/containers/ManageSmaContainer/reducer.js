/*
 *
 * ManageSmaContainer reducer
 *
 */

import { fromJS } from 'immutable';
import * as Constants from './constants';

const initialState = fromJS({
  // mediaServers: [],
  error: false,
  loading: true,
});

function manageSmaContainerReducer(state = initialState, action) {
  switch (action.type) {
    case Constants.GET_MEDIA_SERVERS_REQUEST_SUCCESS:
      // console.log('*****ACTION: mediaServers: ', action.mediaServers);
      return state.set('mediaServers', action.mediaServers).set('loading', false);
    case Constants.GET_MEDIA_SERVERS_REQUEST_FAILURE:
      return state.set('error', `${action.error}`);
    case Constants.GET_MEDIA_SERVERS_REQUEST:
      return state.set('loading', true);
    default:
      return state;
  }
}

export default manageSmaContainerReducer;
