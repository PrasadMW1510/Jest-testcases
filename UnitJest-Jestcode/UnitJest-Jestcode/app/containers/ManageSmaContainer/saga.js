import { all, call, select, put, takeLatest } from 'redux-saga/effects';
import { makeSelectProfileSessionId, makeSelectProfileDistrictId } from 'containers/App/selectors';
import * as Actions from './actions';
import * as Request from './request';
import * as Constants from './constants';

// Individual exports for testing
export function* handleMediaServersRequestFlow() {
  try {
    // const { sessionId, userOrgId } = yield call(getProfileData);
    const sessionId = yield select(makeSelectProfileSessionId());
    const districtId = yield select(makeSelectProfileDistrictId());

    const mediaServers = yield call(Request.getMediaServers, sessionId, districtId);
    yield put(Actions.handleMediaServersRequestSuccess(mediaServers));
  } catch (err) {
    yield put(Actions.handleMediaServersRequestFailure(err));
  }
}

// export function* getProfileData() {
//   const sessionId = yield select(makeSelectProfileSessionId());
//   const userOrgId = yield select(makeSelectProfileUserOrgId());
//   return { sessionId, userOrgId };
// }

export default function* defaultSaga() {
  yield all([takeLatest(Constants.GET_MEDIA_SERVERS_REQUEST, handleMediaServersRequestFlow)]);
}
