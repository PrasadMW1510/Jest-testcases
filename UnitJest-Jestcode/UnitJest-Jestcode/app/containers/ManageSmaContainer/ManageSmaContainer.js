/**
 *
 * ManageSmaContainer
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { compose } from 'redux';

import injectSaga from 'utils/injectSaga';
import injectReducer from 'utils/injectReducer';
import { getMediaServersRequest } from './actions';
import {
  makeSelectManageSmaContainer,
  makeSelectManageSmaContainerGetMediaServers,
} from './selectors';
import reducer from './reducer';
import saga from './saga';
import ManageSma from '../../components/ManageSma/ManageSma';

export class ManageSmaContainer extends React.Component {
  componentDidMount = () => {
    this.props.getMediaServersRequest();
  };

  render() {
    return <ManageSma mediaServers={this.props.mediaServers} />;
  }
}

ManageSmaContainer.defaultProps = {
  mediaServers: [],
};

ManageSmaContainer.propTypes = {
  getMediaServersRequest: PropTypes.func.isRequired,
  mediaServers: PropTypes.array.isRequired,
};

const mapStateToProps = createStructuredSelector({
  manageSmaContainer: makeSelectManageSmaContainer(),
  mediaServers: makeSelectManageSmaContainerGetMediaServers(),
});

const withConnect = connect(mapStateToProps, {
  getMediaServersRequest,
});

const withReducer = injectReducer({ key: 'manageSmaContainer', reducer });
const withSaga = injectSaga({ key: 'manageSmaContainer', saga });

export default compose(withReducer, withSaga, withConnect)(ManageSmaContainer);
