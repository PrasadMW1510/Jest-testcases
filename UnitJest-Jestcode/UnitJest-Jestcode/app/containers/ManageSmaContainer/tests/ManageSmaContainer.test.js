import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { ManageSmaContainer } from '../ManageSmaContainer';

describe('<ManageSmaContainer />', () => {
  let wrapper = null;
  const mockMediaServers = [
    {
      name: ['MMA tests'],
      capabilities: [
        {
          capability: [
            {
              url: ['S3CDN-B8AEED7F1944-107-DaytonStDCNJ.media.education.scholastic.com'],
            },
          ],
        },
      ],
    },
  ];

  const mockGetMediaServersRequest = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <ManageSmaContainer
        getMediaServersRequest={mockGetMediaServersRequest}
        mediaServers={mockMediaServers}
      />
    );
  });

  it('should render correctly', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
