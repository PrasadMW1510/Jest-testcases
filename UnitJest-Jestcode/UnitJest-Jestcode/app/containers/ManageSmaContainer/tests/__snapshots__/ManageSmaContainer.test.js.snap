// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ManageSmaContainer /> should render correctly 1`] = `
<ManageSma
  mediaServers={
    Array [
      Object {
        "capabilities": Array [
          Object {
            "capability": Array [
              Object {
                "url": Array [
                  "S3CDN-B8AEED7F1944-107-DaytonStDCNJ.media.education.scholastic.com",
                ],
              },
            ],
          },
        ],
        "name": Array [
          "MMA tests",
        ],
      },
    ]
  }
/>
`;
